Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp

```diff
@@ -2380,61 +2487,96 @@ static JSC::JSObject* jsResultFromReplyDecoder(JSC::JSGlobalObject* globalObject
         return JSValueMakeUndefined(context);
     }
-    auto connection = processTargetFromArgument(globalObject, arguments[0], exception);
+    auto* globalObject = toJS(context);
+    JSC::JSLockHolder lock(globalObject->vm());
+    auto connection = self->processTargetFromArgument(globalObject, arguments[0], exception);
     if (!connection)
         return JSValueMakeUndefined(context);
-    return sendMessageWithJSArguments(*connection, context, argumentCount - 1, arguments + 1, exception);
+    auto destinationID = destinationIDFromArgument(globalObject, arguments[1], exception);
+    if (!destinationID)
+        return JSValueMakeUndefined(context);
+    auto messageName = messageNameFromArgument(globalObject, arguments[2], exception);
+    if (!messageName)
+        return JSValueMakeUndefined(context);
+    JSValueRef messageArguments = argumentCount > 3 ? arguments[3] : nullptr;
+    return jsSend(connection->connection().get(), *destinationID, *messageName, context, messageArguments, exception);
 }
 
 JSValueRef JSIPC::waitForMessage(JSContextRef context, JSObjectRef, JSObjectRef thisObject, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception)
 {
-    RefPtr jsIPC = toWrapped(context, thisObject);
-    if (!jsIPC) {
+    RefPtr self = toWrapped(context, thisObject);
+    if (!self) {
         *exception = createTypeError(context, "Wrong type"_s);
         return JSValueMakeUndefined(context);
     }
-    if (argumentCount < 3) {
-        *exception = createTypeError(context, "Must specify the target process, destination ID, and message ID as the first three arguments"_s);
+    if (argumentCount < 4) {
+        *exception = createTypeError(context, "Must specify the target process, destination ID, and message ID, timeout as the first four arguments"_s);
         return JSValueMakeUndefined(context);
     }
     auto* globalObject = toJS(context);
     JSC::JSLockHolder lock(globalObject->vm());
-    auto connection = processTargetFromArgument(globalObject, arguments[0], exception);
+    auto connection = self->processTargetFromArgument(globalObject, arguments[0], exception);
     if (!connection)
         return JSValueMakeUndefined(context);
-    return waitForMessageWithJSArguments(*connection, context, argumentCount - 1, arguments + 1, exception);
+    auto info = extractSyncIPCMessageInfo(context, argumentCount - 1, arguments + 1, exception);
+    if (!info)
+        return JSValueMakeUndefined(context);
+    auto [destinationID, messageName, timeout] = *info;
+    return jsWaitForMessage(connection->connection().get(), destinationID, messageName, timeout, context, exception);
 }
 
 JSValueRef JSIPC::sendSyncMessage(JSContextRef context, JSObjectRef, JSObjectRef thisObject, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception)
 {
-    RefPtr jsIPC = toWrapped(context, thisObject);
-    if (!jsIPC) {
+    RefPtr self = toWrapped(context, thisObject);
```

Comment:
I wouldn't call this "self". Maybe "impl" but I think I would have preferred to keep the old name: jsIPC.